src/tablestore.rs

```diff
@@ -270,15 +276,75 @@ impl TableStore {
         handle: &SSTableHandle,
         index: Arc<SsTableIndexOwned>,
         blocks: Range<usize>,
-    ) -> Result<VecDeque<Block>, SlateDBError> {
+        cache_blocks: bool,
+    ) -> Result<VecDeque<Arc<Block>>, SlateDBError> {
         let path = self.path(&handle.id);
         let obj = ReadOnlyObject {
             object_store: self.object_store.clone(),
             path,
         };
-        self.sst_format
-            .read_blocks(&handle.info, index.as_ref(), blocks, &obj)
-            .await
+        let mut blocks_read = VecDeque::with_capacity(blocks.end - blocks.start);
```

Comment:
I *think* I understand this logic. Extensive rustdocs on the `read_blocks_using_index` method would help.

I went back and forth in my head about whether this logic belongs in ReadOnlyObject, SsTableFormat, or here. I think I agree with your implementation that it should be here, especially as we start caching indexes and filters and stuff.